fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineShortcuts.kt

```diff
@@ -0,0 +1,254 @@
+import mozilla.components.browser.state.state.availableSearchEngines
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ContextualMenu
+import org.mozilla.fenix.compose.MenuItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Top-level UI for search shortcuts settings.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun SearchEngineShortcuts(
+    categoryTitle: String,
+    store: BrowserStore,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+    onAddEngineClicked: () -> Unit,
+) {
+    val searchState = store.observeAsComposableState { it.search }.value ?: SearchState()
+    val searchEngines = with(searchState) {
+        regionSearchEngines + additionalSearchEngines + availableSearchEngines + customSearchEngines
+    }
+    val disabledShortcutsIds = searchState.disabledSearchEngineIds
+
+    LazyColumn(
+        modifier = Modifier
+            .background(color = FirefoxTheme.colors.layer1)
+            .fillMaxSize(),
+    ) {
+        item {
+            Title(title = categoryTitle)
+
+            Spacer(modifier = Modifier.height(12.dp))
+        }
+
+        items(
+            items = searchEngines,
+            key = { engine -> engine.id },
+        ) {
+            SearchItem(
+                engine = it,
+                name = it.name,
+                isEnabled = !disabledShortcutsIds.contains(it.id),
+                onCheckboxClicked = onCheckboxClicked,
+                onEditEngineClicked = onEditEngineClicked,
+                onDeleteEngineClicked = onDeleteEngineClicked,
+            )
+        }
+
+        item {
+            AddEngineButton(onAddEngineClicked = onAddEngineClicked)
+        }
+    }
+}
+
+@Composable
+private fun Title(title: String) {
+    Box(
+        modifier = Modifier
+            .height(48.dp)
+            .fillMaxWidth(),
+        contentAlignment = Alignment.CenterStart,
+    ) {
+        Text(
+            text = title,
+            color = FirefoxTheme.colors.textAccent,
+            fontWeight = FontWeight.W400,
+            modifier = Modifier.padding(horizontal = 16.dp),
+            style = FirefoxTheme.typography.headline8,
+        )
+    }
+}
+
+@Suppress("LongParameterList", "LongMethod")
+@Composable
+private fun SearchItem(
+    engine: SearchEngine,
+    name: String,
+    isEnabled: Boolean,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+) {
+    val isMenuExpanded: MutableState<Boolean> = remember { mutableStateOf(false) }
+
+    Row(
+        modifier = Modifier
+            .defaultMinSize(minHeight = 56.dp)
+            .padding(start = 4.dp),
+    ) {
+        Checkbox(
```

Comment:
**Blocker:** I believe the uncheck color for these Checkbox needs to be adjusted for the dark theme. They should appear white for the dark theme.

<img src="https://github.com/mozilla-mobile/firefox-android/assets/92760693/f6547812-8c5f-4d54-876d-e89b9e3c1112" width="300">

Reply:
implemented